src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -473,6 +474,12 @@ internal int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancella
                 return Succeeded;
             }
 
+            if (Arguments.DisplayVersion)
+            {
+                PrintVersion(consoleOutput);
+                return Succeeded;
+            }
```

Comment:
Shouldn't this be the first thing to check? Based on the description of the switch, neither help, nor logo should be printed in presence of /version